fsfreeze.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+)
+
+// from linux/fs.h
+const FIFREEZE = 0xC0045877
+const FITHAW = 0xC0045878
+
+const guestFsfreezeStatusThawed = "thawed"
+const guestFsfreezeStatusFrozen = "frozen"
+
+// Describes the system freeze status.
+var guestFsfreezeStatus string
+
+type MEntry struct {
+	FSSpec string
+	FSFile string
+	FSType string
+}
+
+// fsfreezeFreeze freezes the freezable mounted filesystems.
+func fsfreezeFreeze() (int, error) {
+	frozenFSCount := 0
+	m, err := getMountPoints()
+	if err != nil {
+		return 0, fmt.Errorf("error while getting mountpoints to fsfreeze-freeze: %v", err)
+	}
+
+	for _, mp := range m {
+		fs, err := os.Open(mp.FSFile)
+		if err != nil {
+			return 0, fmt.Errorf("error while opening mount path for fsfreeze-freeze: %v", err)
+		}
+		defer fs.Close()
```

Comment:
not that defer is func-scoped, not block-scoped, so these open file descriptors will pile up during the loop until the end and get all cleaned at once.

that's probably fine, but not ideal

Reply:
Yeah, we discussed about this here: https://github.com/bradfitz/qemu-guest-kragent/pull/3#discussion_r1597382624

To make both you and @stapelberg happy, I've `fs.Close()`d both and the end and in the error branches :)